src/lib.rs

```diff
@@ -1,39 +1,42 @@
-extern crate regex;
-
+use lazy_static::lazy_static;
 use regex::Regex;
 use std::collections::HashMap;
 
-#[macro_use] extern crate lazy_static;
-
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
```

Comment:
was this my mistake from the start, or did something change in rust / best practices?

Reply:
Maybe not quite either? I basically removed things I didn't see used *or* see a specific reason to have in the public API. Here, for example, it wasn't clear what the value of the `PartialOrd` or `Ord` implementations would be to a consumer, and they weren't used internally at all, so the library will be leaner by not having those generated derives. This is tiny, so if you think they *are* valuable to have (they are in the public API!) it’s totally reasonable to leave them in.